File: src/main/java/com/blogly/blogly/domain/user/Name.java
package com.blogly.blogly.domain.user;

import com.blogly.blogly.domain.exception.InvalidNameException;

import java.util.Objects;

public record Name(String value) {

    public Name {
        Objects.requireNonNull(value, "Name value cannot be null");

        if (value.length() < 3 || value.length() > 50) {
            throw new InvalidNameException("Name must be between 3 and 50 characters");
        }
    }
}

File: src/main/java/com/blogly/blogly/domain/user/Name.kt
package com.blogly.blogly.domain.user

@JvmInline
value class Name(val value: String) {
    init {
        require(value.length in MIN_LENGTH..MAX_LENGTH) {
            "Name must be between $MIN_LENGTH and $MAX_LENGTH characters"
        }
    }

    companion object {
        private const val MIN_LENGTH = 3
        private const val MAX_LENGTH = 50
    }
}
